ts/features/fci/screens/valid/FciSignatureFieldsScreen.tsx

```diff
@@ -0,0 +1,255 @@
+import { fciUpdateDocumentSignaturesRequest } from "../../store/actions";
+import { useFciAbortSignatureFlow } from "../../hooks/useFciAbortSignatureFlow";
+import { ClausesTypeEnum } from "../../../../../definitions/fci/ClausesType";
+import { DocumentSignature } from "../../../../../definitions/fci/DocumentSignature";
+import {
+  clausesByType,
+  clauseTypeMaping,
+  getSectionListData
+} from "../../utils/signatureFields";
+
+export type FciSignatureFieldsScreenNavigationParams = Readonly<{
+  documentId: Document["id"];
+  currentDoc: number;
+}>;
+
+const FciSignatureFieldsScreen = (
+  props: IOStackNavigationRouteProps<FciParamsList, "FCI_SIGNATURE_FIELDS">
+) => {
+  const docId = props.route.params.documentId;
+  const currentDoc = props.route.params.currentDoc;
+  const documentsSelector = useSelector(fciSignatureDetailDocumentsSelector);
+  const signatureFieldsSelector = useSelector(
+    fciDocumentSignatureFieldsFieldsSelector(docId)
+  );
+  const documentsSignaturesSelector = useSelector(
+    fciDocumentSignaturesSelector
+  );
+  const dispatch = useIODispatch();
+  const navigation = useNavigation();
+  const [isClausesChecked, setIsClausesChecked] = React.useState(false);
+
+  // get signatureFields for the current document
+  const docSignatures = pipe(
+    documentsSignaturesSelector,
+    RA.findFirst(doc => doc.document_id === docId)
+  );
+
+  React.useEffect(() => {
+    // get required signatureFields for the current document
+    // that user should check to sign the document
+    const requiredFields = clausesByType(
+      signatureFieldsSelector,
+      ClausesTypeEnum.REQUIRED
+    );
+
+    // get required signatureFields for the current document
+    // that user has already checked to sign the document
+    const res = pipe(
+      requiredFields,
+      RA.map(signatureField =>
+        pipe(
+          docSignatures,
+          RA.fromOption,
+          RA.map(doc => doc.signature_fields),
+          RA.map(fields => fields.filter(f => f === signatureField)),
+          RA.flatten
+        )
+      ),
+      RA.flatten
+    );
+
+    setIsClausesChecked(res.length >= requiredFields.length);
+  }, [
+    documentsSignaturesSelector,
+    docId,
+    signatureFieldsSelector,
+    docSignatures
+  ]);
+
+  const { present, bottomSheet: fciAbortSignature } =
+    useFciAbortSignatureFlow();
+
+  const onPressDetail = (signatureField: SignatureField) => {
+    navigation.navigate(FCI_ROUTES.MAIN, {
+      screen: FCI_ROUTES.DOCUMENTS,
+      params: {
+        attrs: signatureField.attrs,
+        currentDoc
+      },
+      merge: true
+    });
+  };
+
+  const updateDocumentSignatures = (fn: (doc: DocumentSignature) => void) =>
+    pipe(
+      docSignatures,
+      O.chain(document => O.fromNullable(document)),
+      O.map(doc => fn(doc))
+    );
+
+  const onChange = (value: boolean, item: SignatureField) =>
+    pipe(
```

Comment:
this is overkill, I'd suggest to use a ternary `value ? doSomething : doSomethingElse`